src/users/users.controller.js

```diff
@@ -0,0 +1,48 @@
+const User = require("../users/users.model");
+const userService = require("./users.service");
+const { CreateUserSchema } = require("../users/users.schema-validator");
+class UserController {
+    async create(req, res) {
+        // check if user exists
+        const existingUser = await userService.findOne({ email: req.body.email })
+        if (existingUser) {
+            return res.status(409).send({ success: false, message: "User already exists" })
+        }
+
+        // create the user
+        const value = await CreateUserSchema.validateAsync(req.body);
```

Comment:
Since we are validating on the route/middleware, there is no need for this line
`UserService.create(req.body)`

Reply:
All reviews as requested are resolved. Please kindly check.